qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -684,3 +712,43 @@ QImage HOCRPdfExporter::getSelection(const QRect& bbox)
 {
 	return m_displayerTool->getSelection(bbox);
 }
+
+void HOCRPdfExporter::paperSizeChanged() {
+	delete ui.lineEditPaperSizeHeight->validator();
+	delete ui.lineEditPaperSizeWidth->validator();
+
+	if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "mm") {
+		ui.lineEditPaperSizeHeight->setValidator(new QIntValidator(0, 10000000));
+		ui.lineEditPaperSizeWidth->setValidator(new QIntValidator(0, 10000000));
+	} else if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "inch") {
+		ui.lineEditPaperSizeHeight->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+		ui.lineEditPaperSizeWidth->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+	}
+
+	auto textPaperSize = ui.comboBoxPaperSize->itemText(ui.comboBoxPaperSize->currentIndex());
+	if(textPaperSize == "Custom") {
+		ui.lineEditPaperSizeWidth->setEnabled(true);
+		ui.lineEditPaperSizeHeight->setEnabled(true);
+	} else if(textPaperSize == "Same as source") {
+		ui.lineEditPaperSizeWidth->setText("");
+		ui.lineEditPaperSizeHeight->setText("");
```

Comment:
I'd store width and height as a setting, and not clear it, even if "same as source" is selected. That way, a custom size the user specifies is kept persistent.

Reply:
I still think these should be stored as settings